src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -1,192 +1,139 @@
+    @Override
+    public void initialize() {
+        frontLeftSpark = new CANSparkMax(RobotMap.CAN.FRONT_LEFT_MOTOR, MotorType.kBrushless);
+        rearLeftSpark = new CANSparkMax(RobotMap.CAN.REAR_LEFT_MOTOR, MotorType.kBrushless);
+        leftSpeedController = new SpeedControllerGroup(frontLeftSpark, rearLeftSpark);
+    
+        frontRightSpark = new CANSparkMax(RobotMap.CAN.FRONT_RIGHT_MOTOR, MotorType.kBrushless);
+        rearRightSpark = new CANSparkMax(RobotMap.CAN.REAR_RIGHT_MOTOR, MotorType.kBrushless);
+        rightSpeedController = new SpeedControllerGroup(frontRightSpark, rearRightSpark);
+        
+        robotDrive = new DifferentialDrive(leftSpeedController, rightSpeedController);                                            
 
-        posController = new SparkPositionControllerGroup(pc_frontLeft, pc_frontRight, pc_rearLeft, pc_rearRight);
-        positiionModeSparks = new FourSparkMaxWithSettings(m_frontLeft, m_rearLeft, m_frontRight, m_rearRight, frontLeftPositionSettings, rearLeftPositionSettings, frontRightPositionSettings, rearRightPositionSettings);
+        frontLeftPositionController = new SparkPositionController(frontLeftSpark, positionSettings);
+        frontRightPositionController = new SparkPositionController(frontRightSpark, positionSettings);
+        rearLeftPositionController = new SparkPositionController(rearLeftSpark, positionSettings);
+        rearRightPositionController = new SparkPositionController(rearRightSpark, positionSettings);
 
-        poseGen = new EncoderPoseGenerator(posController);
-        e_frontLeft = m_frontLeft.getEncoder();
-        e_frontRight = m_frontRight.getEncoder();
-        e_rearLeft = m_rearLeft.getEncoder();
-        e_rearRight = m_rearRight.getEncoder();
+        positionController = new SparkPositionControllerGroup(frontLeftPositionController, 
+                frontRightPositionController, rearLeftPositionController, rearRightPositionController);
 
+        reset();
     }
-
-    @Override 
-    public void periodic(){
-        logger.log("Front Left Encoder Ticks", e_frontLeft.getPosition());
-        logger.log("Front Right Encoder Ticks", e_frontRight.getPosition());
-        logger.log("Rear Left Encoder Ticks", e_rearLeft.getPosition());
-        logger.log("Rear Right Encoder Ticks", e_rearRight.getPosition());
+    public void setSpeedMode(){
+        speedSettings.configureSparkMax(frontLeftSpark);
+        speedSettings.configureSparkMax(frontRightSpark);
+        speedSettings.configureSparkMax(rearLeftSpark);
+        speedSettings.configureSparkMax(rearRightSpark);        
     }
-
-    public void reset(){
-        pc_frontLeft.resetPosition();
-        pc_frontRight.resetPosition();
-        pc_rearLeft.resetPosition();
-        pc_rearRight.resetPosition();
-        logger.log("Clicks per rotation", e_rearRight.getCountsPerRevolution());
+    public void setPositionMode(){
+        positionController.configureAll();
```

Comment:
this seems illogical when i'm looking at setSpeedMode. why do we use a group for the positions, but not for speeds? i think i like the explicit 4-lines much better. IE, i'm realy with mandrews that the positioncontrollergroup isnt adding enough value to be useful.